### ISSUE TITLE
Hang on pg

### ISSUE THREAD

**Issue by akzhan** [will/crystal-pg#98]
> @bcardiff
> From a nasty puts debugging the call to PQ::Connection.read_next_row_start assigns 'E' instead of 'D' probably (?) in type = soc.read_char

> I haven't go any further than this.

```crystal
require "pg"

`psql -c "DROP DATABASE IF EXISTS issue_db_46"`
`createdb issue_db_46`
`psql -d issue_db_46 -c "CREATE TABLE users (name varchar(50), CONSTRAINT key_name UNIQUE(name));"`
`psql -d issue_db_46 -c "INSERT INTO users (name) VALUES ('test');"`

db = DB.open("postgres://localhost/issue-db-46")

puts "A"

db.query "INSERT INTO users (name) VALUES ($1) RETURNING *", "test" do |rs|
  rs.move_next
end

puts "Z"

db.close
```

Related to https://github.com/crystal-lang/crystal-db/issues/46 and https://github.com/Crecto/crecto/issues/76.

**Comment by bigtunacan** [comment]
@will this issue can be closed. It's fixed by PR #100

**Comment by will** [comment]
Thanks!
